test/testBody.cpp

```diff
@@ -1,96 +1,151 @@
-
-	EXPECT_THROW(p.parseHttpRequest("GET /search?query=openai&year=2024#conclusion HTTP/1.1\r\nHost: www.example.com\r\nTransfer-Encoding: gzip, chunked\r\n\r\n6\r\nhello 6\r\nworld!\r\n0\r\n\r\n", request), std::runtime_error);
-    p.clearRequest(request);
-    p.clearParser();
-    try{
-        p.parseHttpRequest("GET /search?query=openai&year=2024#conclusion HTTP/1.1\r\nHost: www.example.com\r\nTransfer-Encoding: gzip, chunked\r\n\r\n6\r\nhello 6\r\nworld!\r\n0\r\n\r\n", request);
-    }
-    catch(const std::runtime_error& e) {
-        std::cerr << e.what() << std::endl;
-    }
+TEST(RequestParser_NonValidBody, MissingCRLFInChunk)
+{
+	RequestParser p;
+	HTTPRequest request;
+
+	EXPECT_THROW(
+		{
+			try {
+				p.parseHeader("POST /search?query=openai&year=2024#conclusion HTTP/1.1\r\nHost: "
+							  "www.example.com\r\nTransfer-Encoding: gzip, chunked\r\n\r\n",
+					request);
+				p.parseBody("6\r\nhello 6\r\nworld!\r\n0\r\n\r\n", request);
+			} catch (const std::runtime_error& e) {
+				EXPECT_STREQ("Invalid HTTP request: Indicated chunk size different than actual chunk size", e.what());
+				throw;
+			}
+		},
+		std::runtime_error);
 }
 
-TEST(RequestParser_NonValidBody, MissingCRInChunk) {
-	RequestParser	p;
-    HTTPRequest request;
-
-	EXPECT_THROW(p.parseHttpRequest("GET /search?query=openai&year=2024#conclusion HTTP/1.1\r\nHost: www.example.com\r\nTransfer-Encoding: gzip, chunked\r\n\r\n6\r\nhello \n6\r\nworld!\r\n0\r\n\r\n", request), std::runtime_error);
-    p.clearRequest(request);
-    p.clearParser();
-    try{
-        p.parseHttpRequest("GET /search?query=openai&year=2024#conclusion HTTP/1.1\r\nHost: www.example.com\r\nTransfer-Encoding: gzip, chunked\r\n\r\n6\r\nhello \n6\r\nworld!\r\n0\r\n\r\n", request);
-    }
-    catch(const std::runtime_error& e) {
-        std::cerr << e.what() << std::endl;
-    }
+TEST(RequestParser_NonValidBody, MissingCRInChunk)
+{
+	RequestParser p;
+	HTTPRequest request;
+
+	EXPECT_THROW(
+		{
+			try {
+				p.parseHeader("POST /search?query=openai&year=2024#conclusion HTTP/1.1\r\nHost: "
+							  "www.example.com\r\nTransfer-Encoding: gzip, chunked\r\n\r\n",
+					request);
+				p.parseBody("6\r\nhello \n6\r\nworld!\r\n0\r\n\r\n", request);
+			} catch (const std::runtime_error& e) {
+				EXPECT_STREQ("Invalid HTTP request: missing CRLF", e.what());
+				throw;
+			}
+		},
+		std::runtime_error);
+}
+
+TEST(RequestParser_NonValidBody, UnexpectedBody)
+{
+	RequestParser p;
+	HTTPRequest request;
+
+	EXPECT_THROW(
+		{
+			try {
+				p.parseHeader("GET /search?query=openai&year=2024#conclusion HTTP/1.1\r\nHost: "
+							  "www.example.com\r\nTransfer-Encoding: gzip, chunked\r\n\r\n",
+					request);
+				p.parseBody("6\r\nhello \n6\r\nworld!\r\n0\r\n\r\n", request);
+			} catch (const std::runtime_error& e) {
+				EXPECT_STREQ("Invalid HTTP request: Method should not have a body", e.what());
+				throw;
+			}
+		},
+		std::runtime_error);
 }
 
 // PREVIOUS TEST CODE
```

Comment:
Since deleted from the other files: could also delete here.
The files are tracked with git, so you could just checkout an older version

Reply:
Yeah, forgot to delete here. Removed.